mlir/lib/Dialect/Rock/Generator/ConvGenerator.cpp

```diff
@@ -654,30 +654,38 @@ ConvGenerator::parseConvDims(int64_t batchSize, int64_t groupSize,
                              int64_t filterHeight, int64_t filterWidth) {
   config.filterDims[DIM::HEIGHT] = filterHeight;
   config.filterDims[DIM::WIDTH] = filterWidth;
-  static const std::string filterKeys = "kgcyx";
```

Comment:
The logic here bothered me.  Keys and Vals are parallel arrays -- given a layout key character, we'll find() it in the Keys array (string) to get an index, and use that index to look up the value in Vals.  Why not just a StringMap to associate the key and value directly?